### ISSUE TITLE
Predicting actions and emotes for the LIGHT Text Adventure Game using the Dodecadialogue model

### ISSUE THREAD

**Issue by bakszero** [facebookresearch/ParlAI#2981]
Using the dodecadialogue pretrained model for the LIGHT dataset (https://parl.ai/docs/zoo.html#light-dialog-models), how is it possible to predict "actions" and "emotes" as described in the paper (https://research.fb.com/wp-content/uploads/2019/11/Learning-to-Speak-and-Act-in-a-Fantasy-Text-Adventure-Game.pdf)?

Note: I was able to set the context as in the issue https://github.com/facebookresearch/ParlAI/issues/2310 here described by @jaseweston the same way as for the bi-ranker model. 

Thanks!

**Comment by klshuster** [comment]
Hi there, you can specify the `--light-label-type` when using `-t light_dialog` to change what the model is meant to predict.

Specifically, if you'd like to see how the model performs (zero-shot) on LIGHT actions, you can run the following command:

```
parlai eval_model -mf zoo:dodecadialogue/light_dialog_ft/model -t light_dialog --light-label-type action
```

See https://github.com/facebookresearch/ParlAI/blob/master/parlai/tasks/light_dialog/agents.py#L102 for all label type choices

**Comment by bakszero** [comment]
@klshuster I think the `eval_model` works, but if I run:

`parlai interactive -mf zoo:dodecadialogue/light_dialog_ft/model -t light_dialog --light-label-type action` or even `parlai interactive -mf zoo:dodecadialogue/light_dialog_ft/model -t light_dialog --light-label-type emote`, it still outputs speech instead of an action or an emote. Not sure if it is possible to get the actions/emotes during interaction with the model?

**Comment by klshuster** [comment]
The `zoo:dodecadialogue/light_dialog_ft/model` was trained to predict the speech label; the `--light-label-type` is only relevant during training to provide the label to the model, but during interactive there isn't currently a way to **force** the generative model to output action/emote (unless you provide some sort of "context" that the model has seen before that indicates it should predict the action/emote). You might want to train with the base model `--init-model zoo:dodecadialogue/base_model/model` on the relevant `--light-label-type` to have a model that can predict actions/emotes (see https://parl.ai/projects/dodecadialogue/ for train settings)

**Comment by klshuster** [comment]
Going to go ahead and close this task as it seems to be resolved, please reopen if there are any further questions
